src/naming/Naming_AST.ml

```diff
@@ -350,6 +350,10 @@ let get_resolved_type lang (vinit, vtype) =
       (* alt: lookup id in env to get its type, which would be cleaner *)
       | Some { e = N (Id (_, { id_type; _ })); _ } -> !id_type
       | Some { e = New (_, tp, _, (_, _, _)); _ } -> Some tp
+      | Some { e = Ref (tok, exp); _ } when lang =*= Go ->
```

Comment:
Why just limit to Go? At the very least could you add a comment explaining that choice?

Reply:
Ah! It was during testing and wanted to limit this to only go: I've tested it with other languages and it works.